CONTRIBUTING.md

```diff
@@ -147,29 +147,61 @@ from the main (upstream) repository:
     git pull --ff upstream master
     ```
     
-## <a name="building-docker"></a>Building a Docker image
-In order to build a Docker image of IBeam, you need to first ensure both Chrome Driver and CPW Gateway are available in `./copy_cache` directory under:
+## <a name="building-docker"></a>Building a Docker Image
 
-* `./copy_cache/chrome_driver` - for the Chrome Driver
-* `./copy_cache/clientportal.gw` - for the Gateway
+The following commands can be used to build the IBeam image after navigating to the root directory of the repository.
 
-You can automate this process by carrying out the following:
+### <a name="local-builds"></a>Local Single-Platform Builds
 
-* Navigate to the root folder of IBeam in your console
-* Run `pip install -r dev-requirements.txt` to install [`invoke` package][invoke]
-* Define the following environment variable paths, contents of which will be copied to the image
-    * `IBEAM_CHROME_DRIVER_PATH` env var pointing at the Chrome Driver directory to be copied from
-    * `IBEAM_GATEWAY_DIR` env var pointing at the Gateway directory to be copied from
-* Make sure `invoke.exe` is in your `$PATH`
-* Run `invoke.exe copyClientportal` and `invoke.exe copyChromeDriver`
-* (optionally) Ensure both `copy_cache/chrome_driver` and `copy_cache/clientportal.gw` are present
+To build the IBeam image for local testing, run:
 
-Finally, run: 
-
-```posh
+```shell
 docker build -t ibeam .
 ```
 
+Alternatively, `docker-compose` can be used to build and run a local IBeam instance using variables in `env.list`:
+
+```shell
+docker-compose up -d --build
+```
+
+### <a name="multi-platform-builds"></a>Multi-Platform Builds
+
+The commands below can be used to setup `docker buildx` for multi-platform IBeam builds supporting `amd64` and `arm64` machines. These commands only need to be executed once per machine and are not required for subsequent multi-platform builds.
+
+Build `docker buildx` from source using:
+
+```shell
+export DOCKER_BUILDKIT=1
+docker build --platform=local -o . git://github.com/docker/buildx
+mkdir -p ~/.docker/cli-plugins
+mv buildx ~/.docker/cli-plugins/docker-buildx
+```
+
+Next, run the following to install `qemu-user-static` for multi-platform build support:
+
+```shell
+docker run --rm --privileged multiarch/qemu-user-static --reset -p yes
+docker buildx create --name builder --driver docker-container --use
+docker buildx inspect --bootstrap
+```
+
+The next two commands can now be used to build a local image for testing (similar to the single-platform builds above) and multi-platform image for pushing to a Docker repository, respectively.
+
+To build and load a single-platform image onto your local machine for testing, run:
+
+```shell
+docker buildx build -t ibeam --load .
+```
+
+To build a multi-platform image and push to a Docker repository, run:
+
+```shell
+docker buildx build --platform linux/amd64,linux/arm64 -t <repo-username>/ibeam:<tag> --push .
```

Comment:
Great stuff 👍 I'd just drop the `--push` flag from this command. Firstly, I don't think others would have permissions to push into `voyz/ibeam` and secondly we'd want the process to involve some testing between building and pushing the image.

Reply:
I agree that they won't be pushing to `voyz/ibeam`, but without this command, the file doesn't really explain how to create a multi-platform build which can be used for testing. Hence, the format is for a generic `repo-username`.

The `--load` flag only builds the image for the building machine's own platform, and will cause `exec error`s on other platforms.

Reply:
Also, wouldn't testing come after the build, possibly in a different section?

Reply:
Hey yeah, so I think we agree here, right? Build, then test, then push. So that would be 

```
docker buildx build --platform linux/amd64,linux/arm64 -t <repo-username>/ibeam:<tag> .
```

Would calling it without `--push` not work?

Reply:
I believe we agree.

My understanding is that with neither `--load` nor `--push` the image is left in the build cache and is not visible, regardless of whether it is multi-platform or not.

When `--load` is used, the image for the building platform is loaded, and visible when running `docker images`. Note that this is only compatible from the platform of the building machine.

When `--push` is used, the multi-platform image is pushed to a repository, and any compatible platform will automatically pull the correct version using `docker pull <image>:<tag>`

Reply:
Ah, I understand what you mean now. Damn, is there any way around it? Can't it be pushed later?

Reply:
Hmm, maybe there's been some miscommunication. If you wanted a build, test, push workflow (and this is referring to the general user who intends to push to their own repo), they would build using `--load`, run their own tests (assuming the tests are run on the building machine's platform), then build using `--push` to deploy to an image repository.

If the tests had to be run on a different platform (e.g. build on `amd64` and test on `arm64`), the order would be build with `--push`, pull the image from the repository and then run tests. Alternatively they could build directly on the testing platform. From testing this is a lot slower on things like a Raspberry Pi 4, may be less of an issue on a Mac.

Reply:
Right that makes sense, thanks for explaining it in more detail. As such, I'd be reluctant to suggest `build -> push -> pull -> test` as the default development cycle. 

Maybe we can outline two workflows: first one for development, where we recommend to use `--load`, and then later when ready to push the one with `--push`?

Reply:
hey @maksimstojkovic what do you recon about the above idea?

Reply:
@Voyz I've separated the build workflows in my latest commit, please have a look.